src/creds.rs

```diff
@@ -259,3 +193,48 @@ fn test_protocol_version() {
     assert_eq!(mls10_e[0], 1);
     assert_eq!(default_e[0], 1);
 }
+
+/// This struct contains a credential and the corresponding private key.
+pub struct CredentialBundle {
+    credential: Credential,
+    signature_private_key: SignaturePrivateKey,
+}
+
+impl CredentialBundle {
+    /// Create a new `CredentialBundle` of the given credential type for the
+    /// given identity and ciphersuite. The corresponding `SignatureKeyPair` is
+    /// freshly generated.
```

Comment:
Generating the signature key pair only makes sense for basic credentials I think. X509 will come with it's own private key. How is this handled?

Reply:
TODO: Create additional constructor that takes X509 reference.

Reply:
I could create a stub for the private counterpart of `Certificate` (which only exists, because it also exists in the spec, but I think we can leave this for issue #134.